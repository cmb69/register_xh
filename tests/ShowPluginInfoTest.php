<?php

/**
 * Copyright (c) 2021 Christoph M. Becker
 *
 * This file is part of Register_XH.
 */

namespace Register;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Register\Infra\DbService;
use Register\Infra\SystemChecker;
use Register\Infra\View;

class ShowPluginInfoTest extends TestCase
{
    public function testRendersPluginInfo()
    {
        $plugin_tx = XH_includeVar("./languages/en.php", 'plugin_tx');
        $lang = $plugin_tx['register'];
        $dbService = $this->createStub(DbService::class);
        $systemChecker = $this->createStub(SystemChecker::class);
        $systemChecker->method('checkVersion')->willReturn(true);
        $systemChecker->method('checkExtension')->willReturn(true);
        $systemChecker->method('checkWritability')->willReturn(true);
        $subject = new ShowPluginInfo("", $lang, $dbService, $systemChecker, new View("./", $lang));

        $response = $subject();

        Approvals::verifyHtml($response);
    }
}
