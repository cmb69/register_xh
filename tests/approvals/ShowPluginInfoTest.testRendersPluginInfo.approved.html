
<h1>Register_XH 2.0-dev</h1>
<div class="register_syscheck">
  <h2>System check</h2>
  <p class="xh_success">Checking that PHP version ≥ 7.0.2 … okay</p>
  <p class="xh_success">Checking that the PHP extension 'hash' is loaded … okay</p>
  <p class="xh_success">Checking that the PHP extension 'session' is loaded … okay</p>
  <p class="xh_success">Checking that CMSimple_XH version ≥ 1.7.0 … okay</p>
  <p class="xh_success">Checking that 'register/css/' is writable … okay</p>
  <p class="xh_success">Checking that 'register/config/' is writable … okay</p>
  <p class="xh_success">Checking that 'register/languages/' is writable … okay</p>
  <p class="xh_success">Checking that '' is writable … okay</p>
</div>
